script.skinshortcuts/resources/lib/datafunctions.py

```diff
@@ -0,0 +1,1339 @@
+
+        # replace unwanted characters
+        text = REPLACE1_REXP.sub('', text.lower()) # replace ' with nothing instead with -
+        text = REPLACE2_REXP.sub('-', text.lower())
+
+        # remove redundant -
+        text = REMOVE_REXP.sub('-', text).strip('-')
+
+        # smart truncate if requested
+        if max_length > 0:
+            text = smart_truncate(text, max_length, word_boundary, '-')
+
+        if separator != '-':
+            text = text.replace('-', separator)
+
+        # If this is a shortcut file (.DATA.xml) and user shortcuts aren't shared, add the skin dir
+        if userShortcuts == True and self.checkIfMenusShared( isSubLevel ) == False:
+            text = "%s-%s" %( xbmc.getSkinDir(), text )
+
+        return text
+
+    # ----------------------------------------------------------------
+    # --- Functions that should get their own module in the future ---
+    # --- (when xml building functions are revamped/simplified) ------
+    # ----------------------------------------------------------------
+
+    def getListProperty( self, onclick ):
+        # For ActivateWindow elements, extract the path property
+        if onclick.startswith( "ActivateWindow" ):
+            # An ActivateWindow - Let's start by removing the 'ActivateWindow(' and the ')'
+            listProperty = onclick
+            # Handle (the not uncommon) situation where the trailing ')' has been forgotten
+            if onclick.endswith( ")" ):
+                listProperty = onclick[ :-1 ]
+            listProperty = listProperty.split( "(", 1 )[ 1 ]
+
+            # Split what we've got left on commas
+            listProperty = listProperty.split( "," )
+
+            # Get the part of the onclick that we're actually interested in
+            if len( listProperty ) == 1:
+                # 'elementWeWant'
+                return listProperty[ 0 ]
+            elif len( listProperty ) == 2 and listProperty[ 1 ].lower().replace( " ", "" ) == "return":
+                # 'elementWeWant' 'return'
+                return listProperty[ 0 ]
+            elif len( listProperty ) == 2:
+                # 'windowToActivate' 'elementWeWant'
+                return listProperty[ 1 ]
+            elif len( listProperty ) == 3:
+                # 'windowToActivate' 'elementWeWant' 'return'
+                return listProperty[ 1 ]
+            else:
+                # Situation we haven't anticipated - log the issue and return original onclick
+                log( "Unable to get 'list' property for shortcut %s" %( onclick ) )
+                return onclick
+        else:
+            # Not an 'ActivateWindow' - return the onclick
+            return onclick
+
+    def upgradeAction( self, action ):
+        # This function looks for actions used in a previous version of Kodi, and upgrades them to the current action
+
+        if not action.lower().startswith("activatewindow("): return action
+
+        # Jarvis + later music windows
```

Comment:
maybe time for a bit of cleanup isn't it? :)

Reply:
@enen92 
He already did. This part is required if the user upgraded from a very old version with old skinshortcuts data. This automatically updates the old actions to valid ones.

Reply:
ok